main.go

```diff
@@ -301,3 +297,31 @@ func loadDaemonFile(cfg *config.Config, flagSet *pflag.FlagSet) error {
 
 	return cfg.MergeConfigurations(flagSet)
 }
+
+// resolveHomeDir resolve a target path from home dir, home dir must not be a relative
+// path, must not be a file, create directory if is not exist, returns the target
+// directory if directory is symlink.
+func resolveHomeDir(path string) (string, error) {
+	if path == "" || !filepath.IsAbs(path) {
+		return "", fmt.Errorf("invalid pouchd's home dir: %s", path)
```

Comment:
I totally disagree with the error message.

If path is empty, the error msg is `invalid pouchd's home dir:`;
if the path is `./a/bc`, the error msg is `invalid pouchd's home dir: ./a/bc`.

Why not split them into two and explain the error reason explicitly.

```
if path == "" {
    return fmt.Errorf("input home-dir cannot be empty")
}
if !path.IsAbs(path){
    return fmt.Errorf("input home-dir %s must be an absolute path", path)
}
```

Otherwise, it would be so hard for user to get noticed what is the exact reason.

WDYT? @Ace-Tang

Reply:
agree.